crates/transport_tx5/src/lib.rs

```diff
@@ -76,11 +87,75 @@ pub mod config {
 
     /// Module-level configuration for Tx5Transport.
     #[derive(Debug, Default, Clone, serde::Serialize, serde::Deserialize)]
+    #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
     #[serde(rename_all = "camelCase")]
     pub struct Tx5TransportModConfig {
         /// Tx5Transport configuration.
         pub tx5_transport: Tx5TransportConfig,
     }
+
+    /// Schema for WebRTC configuration.
+    ///
+    /// This follows this [WebRTC reference](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/RTCPeerConnection#bundlepolicy)
+    /// and may not be up-to-date.
+    ///
+    /// It is also not guaranteed that all options are supported.
+    #[cfg(feature = "schema")]
+    #[allow(dead_code)]
+    fn webrtc_schema(
+        generator: &mut schemars::SchemaGenerator,
+    ) -> schemars::schema::Schema {
+        #[derive(schemars::JsonSchema)]
+        #[schemars(rename_all = "kebab-case")]
+        enum BundlePolicy {
+            Balanced,
+            MaxCompat,
+            MaxBundle,
+        }
+
+        #[derive(schemars::JsonSchema)]
+        #[schemars(rename_all = "camelCase")]
+        struct IceServer {
+            credential: Option<String>,
+            credential_type: Option<String>,
+            urls: Vec<String>,
+            username: Option<String>,
+        }
+
+        #[derive(schemars::JsonSchema)]
+        #[schemars(rename_all = "kebab-case")]
+        enum IceTransportPolicy {
+            All,
+            Public,
+            Relay,
+        }
+
+        #[derive(schemars::JsonSchema)]
+        #[schemars(rename_all = "kebab-case")]
+        enum RtcpMuxPolicy {
+            Negotiate,
+            Require,
+        }
+
+        #[derive(schemars::JsonSchema)]
+        #[schemars(rename_all = "camelCase")]
+        struct WebRtcConfig {
+            bundle_policy: Option<BundlePolicy>,
+            certificates: Option<Vec<serde_json::Value>>,
+            #[schemars(default)]
+            ice_candidate_pool_size: u16,
+            ice_servers: Option<Vec<IceServer>>,
+            ice_transport_policy: Option<IceTransportPolicy>,
+            peer_identity: Option<String>,
+            rtcp_mux_policy: Option<RtcpMuxPolicy>,
+        }
+
+        let root = schemars::schema_for!(WebRtcConfig);
+        for def in root.definitions {
+            generator.definitions_mut().insert(def.0, def.1);
+        }
+        schemars::schema::Schema::Object(root.schema)
```

Comment:
This is how I plan to integrate this into Holochain. Define a schema that mentions our known configuration items and permits unknown ones (see the example) and then just merges it into the parent schema.